File: ts/src/cli/run-cli/resolve-output-path/radius-suffix.ts
/**
 * Stable a-b-c ordering keeps the filename deterministic regardless of
 * CLI argument order. Flags the user did not type are omitted, so an
 * implicit -C 10 default produces no suffix. -C is shorthand for setting
 * both -A and -B; if both are typed explicitly, -C has no remaining
 * effect on the run, so we drop it from the filename too.
 */
export function radiusSuffix(inputs: {
  descendants: number | null;
  ancestors: number | null;
  context: number | null;
}): string {
  const parts: /* mutable */ string[] = [];
  if (inputs.descendants !== null) {
    parts.push(`a${inputs.descendants}`);
  }
  if (inputs.ancestors !== null) {
    parts.push(`b${inputs.ancestors}`);
  }
  const bothExplicit = inputs.descendants !== null && inputs.ancestors !== null;
  if (inputs.context !== null && !bothExplicit) {
    parts.push(`c${inputs.context}`);
  }
  return parts.length === 0 ? "" : `-${parts.join("-")}`;
}

File: crates/unsnarl/src/cli/run_cli/resolve_output_path/radius_suffix.rs
use unsnarl_root_query::GenerationCount;

/// Stable a-b-c ordering keeps the filename deterministic regardless of
/// CLI argument order. Flags the user did not type are omitted, so an
/// implicit -C 10 default produces no suffix. -C is shorthand for setting
/// both -A and -B; if both are typed explicitly, -C has no remaining
/// effect on the run, so we drop it from the filename too.
pub fn radius_suffix(
    descendants: Option<GenerationCount>,
    ancestors: Option<GenerationCount>,
    context: Option<GenerationCount>,
) -> String {
    let mut parts: Vec<String> = Vec::new();
    if let Some(n) = descendants {
        parts.push(format!("a{}", n.0));
    }
    if let Some(n) = ancestors {
        parts.push(format!("b{}", n.0));
    }
    let both_explicit = descendants.is_some() && ancestors.is_some();
    if let Some(n) = context {
        if !both_explicit {
            parts.push(format!("c{}", n.0));
        }
    }
    if parts.is_empty() {
        String::new()
    } else {
        format!("-{}", parts.join("-"))
    }
}

#[cfg(test)]
#[path = "radius_suffix_test.rs"]
mod radius_suffix_test;
